tests/js/uuid-sync-tests.js

```diff
@@ -0,0 +1,131 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+"use strict";
+
+const Realm = require("realm");
+const TestCase = require("./asserts");
+const AppConfig = require("./support/testConfig");
+
+const {UUID} = Realm.BSON;
+
+const SCHEMA = {
+    name: "UUIDObject",
+    primaryKey: "_id",
+    properties: {
+        _id: "uuid",
+        mandatory: "uuid",
+        optional: "uuid?",
+        list: "uuid[]"
+    }
+}
+
+module.exports = {
+    async testUUIDSync() {
+        if (!global.enableSyncTests) return;
+
+        const appConfig = AppConfig.integrationAppConfig;
+        const app = new Realm.App(appConfig);
+        const credentials = Realm.Credentials.anonymous();
+
+        const user = await app.logIn(credentials);
+        const config = {
+            sync: {
+                user,
+                partitionValue: "UUID",
```

Comment:
Could this partitionValue just be any string?  "UUID" has no special meaning here, correct?

Reply:
correct, it just seemed to fit the theme :)